bin/DESeq2.R

```diff
@@ -41,22 +49,28 @@ dir.create("differential_gene_expression/final_gene_table")
 # check input data path
 # provide these files as arguments:
 option_list = list(
+    make_option(c("-y", "--input_type"), type="character", default="rawcounts", help="Which type of input data is provided; must be one of [rawcounts, rsem, salmon]", metavar="character"),
     make_option(c("-c", "--counts"), type="character", default=NULL, help="path to raw count table", metavar="character"),
     make_option(c("-m", "--metadata"), type="character", default=NULL, help="path to metadata table", metavar="character"),
     make_option(c("-d", "--design"), type="character", default=NULL, help="path to linear model design file", metavar="character"),
     make_option(c("-x", "--contrasts_matrix"), type="character", default=NULL, help="path to contrasts matrix file", metavar="character"),
+    make_option(c("-f", "--gtf"), type="character", default=NULL, help="path to gtf table if using salmon/rsem input", metavar="character"),
     make_option(c("-r", "--relevel"), type="character", default=NULL, help="path to factor relevel file", metavar="character"),
     make_option(c("-k", "--contrasts_list"), type="character", default=NULL, help="path to contrasts list file", metavar="character"),
     make_option(c("-p", "--contrasts_pairs"), type="character", default=NULL, help="path to contrasts pairs file", metavar="character"),
     make_option(c("-l", "--genelist"), type="character", default=NULL, help="path to gene list file", metavar="character"),
     make_option(c("-t", "--logFCthreshold"), type="integer", default=0, help="Log 2 Fold Change threshold for DE genes", metavar="character"),
     make_option(c("-g", "--rlog"), type="logical", default=TRUE, help="if TRUE, perform rlog transformation", metavar="character"),
-    make_option(c("-n", "--vst_genes_number"), type="integer", default=1000, help="subset number of genes for vst", metavar="character"),
     make_option(c("-b", "--batchEffect"), default=FALSE, action="store_true", help="Whether to consider batch effects in the DESeq2 analysis", metavar="character")
 )
-
 opt_parser = OptionParser(option_list=option_list)
 opt = parse_args(opt_parser)
+if (!(opt$input_type %in% c("rawcounts", "rsem", "salmon"))){
+    stop(paste0("Wrong input type ", opt$input_type, ", must be one of [rawcounts, rsem, salmon]!"))
+}
+if (opt$input_type %in% c("rsem", "salmon") && is.null(opt$gtf)){
+    stop(paste0("For input type salmon, gtf file needs to be provided!"))
```

Comment:
```suggestion
    stop(paste0("For input type salmon, gtf file needs to be provided!\nIf using igenomes, please check that the entry for your genome contains a gtf file and otherwise provide one with `--gtf`."))
```